### PR TITLE
fix post install method for py3 in mercurial

### PR DESCRIPTION
**Pull request by healther**
make mercurial/package.py python3 compatible

### CODE REVIEW

**Comment by healther** [comment]
how would I use this here? A quick search in the documentation didn't turn up anything useful

**Comment by adamjstewart** [comment]
Hmm, try `spec['python'].print_string`? @alalazo might know more.

**Comment by healther** [comment]
It's ```self.spac['python'].package.print_string()``` but that doesn't solve this problem as we need to import the ```certifi``` module first

File: var/spack/repos/builtin/packages/mercurial/package.py [hunk 1]
```diff
@@ -47,7 +48,9 @@ class Mercurial(PythonPackage):
     version('3.8.2', 'c38daa0cbe264fc621dc3bb05933b0b3')
     version('3.8.1', '172a8c588adca12308c2aca16608d7f4')
 
-    depends_on('python@2.6:2.8')
+    depends_on('python')
+    depends_on('python@2.6:2.8', when='@:4.2.99')
+    conflicts('python@3.0:3.4')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mercurial/package.py]: This shouldn't be in `conflicts`, it should be in `depends_on`.

Comment (healther) [reply var/spack/repos/builtin/packages/mercurial/package.py]: Done but I don't quite get the idea of conflicts() then. My reasoning was:
- In general we depend on python (i.e. all versions need some form of python) => `depends_on('python')`
- If the version is below 4.3 python 2 is required => `depends_on('python@2.6:2.8')`
- There is no support for python 3 below 3.5 => `conflicts('python@3.0:3.4')`

File: var/spack/repos/builtin/packages/mercurial/package.py [hunk 2]
```diff
@@ -47,7 +48,8 @@ class Mercurial(PythonPackage):
     version('3.8.2', 'c38daa0cbe264fc621dc3bb05933b0b3')
     version('3.8.1', '172a8c588adca12308c2aca16608d7f4')
 
-    depends_on('python@2.6:2.8')
+    depends_on('python@2.6:2.8', when='@:4.2.99')
+    depends_on('python@3.5:', when='@4.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mercurial/package.py]: It looks like 4.4.1 still supports Python 2.7:
```python
    # Mercurial will never work on Python 3 before 3.5 due to a lack            
    # of % formatting on bytestrings, and can't work on 3.6.0 or 3.6.1          
    # due to a bug in % formatting in bytestrings.                              
    #                                                                           
    # TODO: when we actually work on Python 3, use this string as the           
    # actual supportedpy string.                                                
    supportedpy = ','.join([                                                    
        '>=2.7',                                                                
        '!=3.0.*',                                                              
        '!=3.1.*',                                                              
        '!=3.2.*',                                                              
        '!=3.3.*',                                                              
        '!=3.4.*',                                                              
        '!=3.6.0',                                                              
        '!=3.6.1',                                                              
    ])                                                                          
```
I think you want `python@2.7:2.8,3.5:3.5.999,3.6.2:`

File: var/spack/repos/builtin/packages/mercurial/package.py [hunk 3]
```diff
@@ -47,7 +48,8 @@ class Mercurial(PythonPackage):
     version('3.8.2', 'c38daa0cbe264fc621dc3bb05933b0b3')
     version('3.8.1', '172a8c588adca12308c2aca16608d7f4')
 
-    depends_on('python@2.6:2.8')
+    depends_on('python@2.6:2.8', when='@:4.2.99')
+    depends_on('python@2.7:2.8,3.5:3.5.999,3.6.2:', when='@4.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mercurial/package.py]: What about Mercurial 4.3?

Comment (healther) [reply var/spack/repos/builtin/packages/mercurial/package.py]: I really don't know. I can add a comment, but if someone adds that she would probably know that.

**Comment by healther** [comment]
@adamjstewart any further comments?

**Comment by adamjstewart** [comment]
Nope, looks good to me. I'm assuming you figured out exactly which versions of Mercurial support which versions of Python. I'll let someone else approve and merge.

**Comment by healther** [comment]
@junghans would you care to take a look?

**Comment by healther** [comment]
@junghans @alalazo anyone? pretty please?

**Comment by adamjstewart** [comment]
I'll just merge it. Sorry you've had to wait this long to get it merged.

**Comment by scheibelp** [comment]
@healther you can also ping me on PRs that you are waiting on

**Pull request merged**